src/modules/item/sagas.ts

```diff
@@ -142,7 +150,33 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
 
   function* handleFetchRaritiesRequest() {
     try {
-      const rarities: Rarity[] = yield call([legacyBuilder, 'fetchRarities'])
+      const rarities: Rarity[] = yield call([legacyBuilder, legacyBuilder.fetchRarities])
```

Comment:
Just out of curiosity, could we move this logic to the backend and make just 1 call using the `currency` that we need to fetch? (`{currency: 'USD'}` if the env flag is one). I'm asking because is not clear why we need to fetch both endpoints and consolidate the data after that.

Reply:
I need both the price in USD and in MANA in the UI because according to the designs, I need to display the total cost in USD as well as in MANA.

At first I did it like this because I didn't want to change the type returned by the backend. depending on the flag, the only difference would be the price. Still, I ended up adding the currency field to the response.

I'll check out if I can convert it to a single endpoint while keeping compatibility.

Reply:
Updated so now only 1 request is made to the server.